demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php

```diff
@@ -3019,8 +3019,11 @@ private function attachSimilarStatementSubmitters(Statement $statementToAttachTo
     public function createNonOriginalStatement(array $originalStatementData, Statement $newOriginalStatement): Statement
     {
         $fieldsForUpdateStatement = $this->extractFieldsForUpdateStatement($originalStatementData);
-        $copyOfStatement = $this->statementCopier->copyStatementObjectWithinProcedureWithRelatedFiles($newOriginalStatement, false, true);
-
+        if ($newOriginalStatement->getFiles()) {
```

Comment:
Wouldn't it be better to have this logic within `copyStatementObjectWithinProcedureWithRelatedFiles`? It should be possible to check there whether files are set or not

Reply:
done in [1d8d5cb](https://github.com/demos-europe/demosplan-core/pull/2515/commits/1d8d5cb490dd5c96aa187f962b7a6e7b7edc2ae1)